src/coreclr/debug/runtimeinfo/datadescriptor.cpp

```diff
@@ -123,6 +138,15 @@ enum
 #include "datadescriptor.h"
 };
 
+// count the global strings
+enum
+{
+    CDacBlobGlobalStringsCount =
+#define CDAC_GLOBALS_BEGIN() 0
+#define CDAC_GLOBAL_STRING(name,value) + 1
+#include "datadescriptor.h"
+};
```

Comment:
I'd like to see explicit #undef 's of these macros after use, because otherwise we're implicitly relying on their presence the next time we include datadescriptor.h further down

Reply:
I see datadescriptor undefs them after it's included, so that's fine then. Just magical.